File: 2021/test/day_01_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_01"

class Day01Test < Minitest::Test
  def test_part01
    day01 = Day01.new

    assert_equal 7, day01.part1(input("day_01_example_1"))
    assert_equal 1602, day01.part1(input("day_01"))
  end

  def test_part02
    day01 = Day01.new

    assert_equal 5, day01.part2(input("day_01_example_1"))
    assert_equal 1633, day01.part2(input("day_01"))
  end
end

File: 2025/test/day_01_test.exs
defmodule Day01Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day01.part_1(input("day_01_example_1")) == 3
    assert Day01.part_1(input("day_01")) == 1139
  end

  test "part2" do
    assert Day01.part_2(input("day_01_example_1")) == 6
    assert Day01.part_2(input("day_01")) == 6684
  end
end
